src/components/ui/UITypography/UITypography.tsx

```diff
@@ -1,27 +1,145 @@
-import { Typography, ThemeProvider, createTheme } from '@mui/material';
-import React from 'react';
+import { ThemeProvider, Typography, createTheme } from '@mui/material';
 
 import { UITypographyProps } from './UITypographyTypes';
 
-const defaultProps: UITypographyProps = {
-  children: '',
+const fonts = {
+  golos: 'Golos Text',
+  inter: 'Inter',
+};
+
+const hStyles = {
+  color: '#1A1C1E',
+  fontFamily: fonts.golos,
+  fontWeight: '700',
+};
+
+const mediumStyles = {
+  color: '#484848',
+  lineHeight: '18px',
+};
+
+const textStyles = {
+  lineHeight: 'normal',
 };
 
 const theme = createTheme({
+  typography: {
+    h1: {
+      ...hStyles,
+      fontSize: '56px',
+    },
+    h2: {
+      ...hStyles,
+      fontSize: '46px',
+    },
+    h3: {
+      ...hStyles,
+      fontSize: '36px',
+      fontWeight: '600',
+    },
+    h4: {
+      ...hStyles,
+      color: '#484848',
+      fontSize: '30px',
+      fontWeight: '600',
+    },
+    h5: {
+      ...hStyles,
+      fontSize: '28px',
+    },
+    h6: {
+      ...hStyles,
+      fontSize: '22px',
+    },
+    medium16: {
+      ...mediumStyles,
+      fontFamily: fonts.inter,
+      fontSize: '16px',
+      fontWeight: '500',
+    },
+    medium15: {
+      ...mediumStyles,
+      fontFamily: fonts.golos,
+      fontStyle: 'normal',
+      fontSize: '15px',
+      color: '#57595B',
+    },
+    medium14: {
+      ...mediumStyles,
+      fontFamily: fonts.inter,
+      fontSize: '14px',
+      color: '#404142',
+    },
+    regular16: {
+      ...mediumStyles,
+      fontFamily: fonts.inter,
+      fontWeight: '400',
+    },
+    bodyText18: {
+      ...textStyles,
+      fontFamily: fonts.golos,
+      fontSize: '18px',
+      fontWeight: '400',
+      lineHeight: '30px',
+    },
+    bodyText16: {
+      ...textStyles,
+      fontFamily: fonts.golos,
+      fontSize: '16px',
+      lineHeight: '26px',
+    },
+    bold22: {
+      ...textStyles,
+      fontFamily: fonts.golos,
+      fontSize: '22px',
+      fontWeight: '700',
+      color: '#57595B',
+    },
+    demi18: {
+      ...textStyles,
+      fontFamily: fonts.golos,
+      fontSize: '18px',
+      fontWeight: '600',
+    },
+    button: {
+      color: '#FFF',
+      fontFamily: fonts.golos,
+      fontSize: '18px',
+      fontStyle: 'normal',
+      fontWeight: '600',
+      lineHeight: 'normal',
+    },
+  },
   components: {
     MuiTypography: {
-      defaultProps,
-      variants: [],
+      defaultProps: {
+        variantMapping: {
+          medium16: 'p',
+          medium15: 'p',
+          medium14: 'p',
+          regular16: 'p',
+          bodyText18: 'p',
+          bodyText16: 'p',
+          bold22: 'p',
+          demi18: 'p',
+          button: 'p',
+        },
+      },
     },
   },
 });
 
-function UITypography({ children }: UITypographyProps) {
+export default function UITypography({
```

Comment:
component from MUI can include a lot of other properties, need give possibility to transfer other properties

Reply:
Done